gemma-rest/src/main/java/ubic/gemma/rest/DatasetsWebService.java

```diff
@@ -1054,6 +1058,48 @@ public ResponseDataObject<List<ExperimentExpressionLevelsValueObject>> getDatase
         );
     }
 
+    /**
+     * Obtain differential expression results for a given gene.
+     */
+    @GET
+    @Path("/differential/genes/{gene}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Retrieve the differential expression results for a given gene")
+    public ResponseDataObject<List<DifferentialExpressionAnalysisResultWithDatasetIdValueObject>> getDatasetsDifferentialExpressionForGene(
+            @PathParam("gene") GeneArg<?> geneArg,
+            @QueryParam("query") QueryArg query,
+            @QueryParam("filter") @DefaultValue("") FilterArg<ExpressionExperiment> filter
+    ) {
+        Gene gene = geneArgService.getEntity( geneArg );
+        Set<Long> ids = new HashSet<>( expressionExperimentService.loadIdsWithCache( datasetArgService.getFilters( filter ), null ) );
+        if ( query != null ) {
+            ids.retainAll( datasetArgService.getIdsForSearchQuery( query ) );
+        }
+        if ( ids.isEmpty() ) {
+            Responder.respond( Collections.emptyList() );
+        }
+        Map<DifferentialExpressionAnalysisResult, Long> datasetByResult = new HashMap<>();
+        List<DifferentialExpressionAnalysisResultWithDatasetIdValueObject> rs = differentialExpressionAnalysisService.findResultsByGene( gene, ids, datasetByResult ).stream()
+                .map( r -> new DifferentialExpressionAnalysisResultWithDatasetIdValueObject( r, datasetByResult.get( r ) ) )
+                .collect( Collectors.toList() );
+        // TODO: pick one result per gene and dataset
```

Comment:
Some genes will have more than one probe associated in some datasets (mainly microarray), so we have to pick the "best" one.